internal/provider/resource_certificate_authority.go

```diff
@@ -0,0 +1,277 @@
+/*
+ *  Copyright 2023 VMware, Inc.
+ *    SPDX-License-Identifier: MPL-2.0
+ */
+
+package provider
+
+import (
+	"context"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/vmware/terraform-provider-vcf/internal/api_client"
+	"github.com/vmware/terraform-provider-vcf/internal/constants"
+	"github.com/vmware/vcf-sdk-go/client/certificates"
+	"github.com/vmware/vcf-sdk-go/models"
+	"time"
+)
+
+func getSupportedCertificateAuthorityTypes() []string {
+	return []string{"Microsoft", "OpenSSL"}
+}
+
+func ResourceCertificateAuthority() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceCertificateAuthorityCreate,
+		ReadContext:   resourceCertificateAuthorityRead,
+		UpdateContext: resourceCertificateAuthorityUpdate,
+		DeleteContext: resourceCertificateAuthorityDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Timeouts: &schema.ResourceTimeout{
```

Comment:
Long timeouts not required for these resources. They are very quick to create. You can make it same as the ceip resource above - 30 mins.